utils.go

```diff
@@ -26,7 +28,7 @@ func countSetBits(data []byte) int {
 func countCommonPrefix(data1 []byte, data2 []byte) int {
 	count := 0
 	for i := 0; i < len(data1)*8; i++ {
-		if hasBit(data1, i) == hasBit(data2, i) {
+		if getBitAtFromMSB(data1, i) == getBitAtFromMSB(data2, i) {
```

Comment:
Similarly, we can use [bits.LeadingZeros8](https://golang.org/pkg/math/bits/#LeadingZeros8) on XORed bytes of `data1` and `data2`.

Reply:
This is definitely a good optimization (could reduce cost of `countCommonPrefix` up to 8-fold), but outside the scope of this PR.